python/paddle/tensor/einsum.py

```diff
@@ -0,0 +1,909 @@
+# Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import itertools
+import re
+
+from ..fluid.layers import reshape, transpose
+from .linalg import matmul
+from .manipulation import squeeze, unsqueeze
+from .math import multiply
+from .math import sum as paddle_sum
+
+from paddle.common_ops_import import dygraph_only
+
+__all__ = []
```

Comment:
Specify the `einsum` as the public API. The others are just  function helpers for `einsum`

Reply:
Should I just leave it as it is since it's exported in paddle/__init__.py ?
Not very sure...

Reply:
OK, it makes senses since it's exported in __init__.py.